### PR TITLE
Remove csproj.md file

### LINKED ISSUE #18198: Remove csproj.md file
Extracted from #2642. Add a redirect to https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props. Also update any AKA links that might be linking there.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #18198, #14683.

Content was migrated to https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props, https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets, and new [assembly info file](https://review.docs.microsoft.com/en-us/dotnet/core/migration/assembly-info?branch=master).

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
Looks good, a couple of questions and comments to consider.

File: docs/core/migration/assembly-info.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/core/migration/assembly-info.md:1]: The file is mostly aimed at customers moving from .NET Core 2.0; for example, the first column in the table is the old attribute from the AssemblyInfo file. Also, I think these properties are mostly NuGet package properties, for which the source of truth is here in the NuGet docs: https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#pack-target.

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -510,6 +510,51 @@ The project file snippet in the following example references the [System.Runtime
...
+- `Analyzers` – the analyzers are used.
+
+Alternatively, the attribute can contain:
+
+- `None` – none of the assets are used.
+- `All` – all assets are used.
+
+### PackageTargetFallback
+
+The `PackageTargetFallback` property lets you specify a set of compatible targets to be used when restoring packages. It's designed to allow packages that use the `dotnet` [Target x Moniker (TxM)](/nuget/reference/target-frameworks) to operate with packages that don't declare a `dotnet` TxM. If your project uses the `dotnet` TxM, then, unless you add `PackageTargetFallback`, all the packages it depends on must also have a dotnet TxM. `PackageTargetFallback` allows non-`dotnet` platforms to be compatible with `dotnet`.
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/core/project-sdk/msbuild-props.md:540]: Turns out this property [is deprecated](#2816).

**Comment by gewarren** [comment]
Thank you Tom!

**Pull request merged**